app/models/external_upload_stub.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require "digest/sha1"
+
+class ExternalUploadStub < ActiveRecord::Base
+  CREATED_EXPIRY_HOURS = 1
+  UPLOADED_EXPIRY_HOURS = 24
+
+  belongs_to :created_by, class_name: 'User'
+
+  scope :expired_created, -> {
+    where(
+      "status = ? AND created_at <= ?",
+      ExternalUploadStub.statuses[:created],
+      CREATED_EXPIRY_HOURS.hours.ago
+    )
+  }
+
+  scope :expired_uploaded, -> {
+    where(
+      "status = ? AND created_at <= ?",
+      ExternalUploadStub.statuses[:uploaded],
+      UPLOADED_EXPIRY_HOURS.hours.ago
+    )
+  }
+
+  before_create do
+    self.unique_identifier = SecureRandom.hex(20)
+    self.status = ExternalUploadStub.statuses[:created] if self.status.blank?
+  end
+
+  def self.statuses
+    @statuses ||= Enum.new(
+      created: 1,
+      uploaded: 2,
+      failed: 3
+    )
+  end
+
+  # TODO (martin): Lifecycle rule would be best to clean stuff up in the external
+  # systems, I don't think we really want to be calling out to the external systems
+  # here right?
+  def self.cleanup!
+    expired_created.destroy_all
+    expired_uploaded.destroy_all
```

Comment:
Perhaps just use `delete_all` here? No need to create ActiveRecord objects just to delete records from the database.